### DISCUSSION TITLE
Scarf joint seams

### DISCUSSION THREAD

**Discussion by tsmith35** [OrcaSlicer/OrcaSlicer#4325] [General]
With scarf joint seams now being beta tested in OrcaSlicer, there are many interested people who want to try out this feature.

This is just a place to discuss how to use scarf seam joints, the handful of settings that affect scarf seam joint quality and any issues found with this new and innovative seam type.

Here is the original discussion, for reference: [Seam: use scarf joint to minimize seam visiblity](https://github.com/SoftFever/OrcaSlicer/pull/3839)

**Comment by psiberfunk** [comment]
Great idea.  Thanks.  I just updated the printables page w/ new builds and[ instructions to answer the most common questions here](https://www.printables.com/model/783313-better-seams-an-orca-slicer-guide-to-using-scarf-s).

**Comment by vgdh** [comment]
It could be used with first layer also if it thick enough. With some start height in order to safe bed from scratching.

**Comment by Tawpie** [comment]
Bit of a mystery... build #1097 (MacOS 14.3.1 using the arm64 .dmg). Two simple objects, my intention was to try to isolate what sounded like dragging on the top of the model but when sliced, 5g of model is expected to take a bit over an hour to print. Turns out it's largely because the inner walls are bring printed desperately slowly. The left model is sliced with contour+hole, all around, 5 scarf steps. The right model is no scarf seams as a reference.

EXPECTED: inner walls print closer to 300 mm/sec (per speed settings and MVF)
ACTUAL: inner walls printing at an average 1.6 seconds/mm

![image](https://github.com/SoftFever/OrcaSlicer/assets/7697076/399a0629-e016-46aa-825e-98a8bd17e4fe)
![image](https://github.com/SoftFever/OrcaSlicer/assets/7697076/190f7dea-d0c9-4f76-b6ca-3fef9d66abb8)

This is the .3mf. Not sure if this is intentional, but it's surprising!
[dragging-1097.3mf.zip](https://github.com/SoftFever/OrcaSlicer/files/14505405/dragging-1097.3mf.zip)

**Comment by khenderick** [comment]
Today I used scarf joints for the first time, and it's awesome, thanks a lot! 

I noticed that the first layer's scarf joint looks a bit odd. It looks like the starting ramp is a bit too thin and/or has troubles properly sticking to the build plate. The end of the scarf looks good, and from the 2nd layer up it's just near perfect.

**Reply by T4psi** [reply]
I noticed this too. On my model, the scarf seam is not continous and every time a new seam line start, there is some underextrusion(?) or other odd phenomenom going on that layer. I've tried adjusting scarf joint speed, extrusion rate smoothing, scarf flow, scarft length and what else, but I could not remove this defect.
<img width="411" alt="image" src="https://github.com/SoftFever/OrcaSlicer/assets/54436826/66d44b37-4857-4a1a-b45e-9ac0bb5867d9">

**Reply by T4psi** [reply]
After investigating the Printables page, I have to add that I was printing 300 mm/s inner wall and 200 mm/s outer wall. Now I see that this is a no-go.

**Reply by psiberfunk** [reply]
It’s possible with very well tuned settings , but you’re likely pushing your luck with this

**Reply by vgdh** [reply]
try start height 0.1 and seam gap = 0

**Comment by vgdh** [comment]
It would be truly great if our scarf settings could consider the percentage of overhang. Imagine, as we print a funnel, having the scarf applied on the inside where there's no overhang, while elegantly abstaining from application on the exterior.

**Comment by MadGoatHaz** [comment]
need arc incorporated... Can clearly see steps in the joint. very much enjoy using them however.

**Reply by Tikonderoga** [reply]
Seems like having scarf joints enabled disables arc fitting completely.

**Comment by darco415** [comment]
what the difference between contour and hole and contour?

**Reply by psiberfunk** [reply]
This is clearly described in the printables post .  Please to read : https://www.printables.com/model/783313-better-seams-an-orca-slicer-guide-to-using-scarf-s

**Reply by theophile** [reply]
As far as I can tell the Printables post just says: 

> Use of **Contour** mode will give you scarf seams on the outermost wall only.  **Contour+Hole** will include inside “hole” walls in the part.  One mode vs the other may be desirable for different reasons.  My settings use Contour+Hole by default.

I don't know what a "hole wall" is. I sliced a model in both Contour and Coutour+Hole mode and couldn't tell what was different between them. Can anyone elaborate?

**Reply by psiberfunk** [reply]
A “hole” is any interior perimeter that is not the outermost wall in a model .  A typical example would be a screw hole , for example

**Reply by theophile** [reply]
Okay I think I get what you mean. I was reading "outermost wall" to mean the outer loop among the set of loops constituting a wall. And I was reading "Contour+Hole will include inside “hole” walls in the part" to mean that this setting will add something called a "hole wall" to the interior of the part somewhere. But I think what you're saying is that "Contour" will only use scarf joints on the exterior shell of the model whereas "Contour+Hole" will use scarf joints even on interior walls. That makes sense!

**Reply by psiberfunk** [reply]
correct.  That's the right way to think about it.  Once you play w/ a few models you can quickly see how it works.

**Comment by wanderlust-vlad** [comment]
Hi there!
I've tried the scarfseams feature in the 2.0.0 beta.
In my case some layer lines bump out of the model. It is even reflected in the slicer:
![image](https://github.com/SoftFever/OrcaSlicer/assets/128132691/8ff0e53a-cac2-4f44-a251-dc9289ff0eea)
![image](https://github.com/SoftFever/OrcaSlicer/assets/128132691/7f217905-40a6-423f-a5ca-055415f6d1fb)
![image](https://github.com/SoftFever/OrcaSlicer/assets/128132691/c578bf35-3528-404b-a4c8-f1e45223add2)
![image](https://github.com/SoftFever/OrcaSlicer/assets/128132691/f45afac1-e9d0-4b0f-b304-6317d7de8e9f)

Scarf settings are default: 100% speed (60 mm/s in my case), 0 start height, 20 mm length, 10 steps. Have I done smth wrong?

**Reply by vgdh** [reply]
Try to use the whole perimeter scarf and start height 0.1

**Reply by psiberfunk** [reply]
Another possible cause of this is actually inner wall #/size  changes @vgdh .  My testing shows that classic mode can help with this sort of thing when arachne would cause such bumps.. but I only saw this effect strongly on overhangs where scarfs were already struggling.

Also, it looks like he isnt using recommended aligned seams..    I wonder what his results would be with the printables profile ?

**Comment by Goodwine** [comment]
I am trying this model out [tardigrade, makerworld](https://makerworld.com/en/models/153107). I use it as a swatch and quick stringing test whenever I use a new filament. Using X1C, 6mm nozzle and calibrated filaments (flow dynamics and rate). My prints are getting holes in them, and I'm not sure what to configure to avoid that.

Attached below is a screenshot of the slice view, and photos of the results in (gold) PLA Silk and (blue) PLA. The issue is only apparent on PLA Silk, so I'm not sure if the settings are incorrect. 

![orca slicer view of tardigrade](https://github.com/SoftFever/OrcaSlicer/assets/2022649/3894fb59-53fb-4b8f-8d3c-98d38ab01c53)

![blue PLA print with no issues](https://github.com/SoftFever/OrcaSlicer/assets/2022649/3d1ff164-ef60-4821-b558-3239fd92e78d)

![gold PLA Silk print with issues](https://github.com/SoftFever/OrcaSlicer/assets/2022649/837f3ea2-e55d-4e9e-b8a5-01549f7f93ca)

**Reply by Goodwine** [reply]
Scarf settings are the default for `Contour`

**Reply by Nicomtl** [reply]
I am getting similar blobs with PETG. It was not happening with PLA. I am working through some settings now to how to correct it.

**Comment by skl111** [comment]
How do I use scarf Joint and have add more flow

> ome underextrusion(?) or other odd phenomenom going on that layer. I've tried adjusting scarf joint speed, extrusion rate smoothing,

Where can you adjust the scarf joint flow, as that is causing under extrusion for me printing some cylindrical models as I can see in the slicer it slows the flow way down from 4 to .01m but I don't see a way co change it.
Using the latest main nightly build from 24-04-28 as o 1030EST. It has been and issue for a while and I finally tracked it down to what setting I was changed that introduced it.

![Screenshot 2024-04-28 224627](https://github.com/SoftFever/OrcaSlicer/assets/25110469/62e797a9-8342-4308-86dd-06a6cdbed3f2)

**Comment by evilDave** [comment]
I was going to open a new issue but thought I might post here first: is anyone else completely unable to get a good result from scarf seams? I feel like I have my printer pretty well dialled in, linear advance, acceleration, speed limits, retraction, max flow rate, etc. set correctly. And it does not seem like it's able to do scarf joint at all with any kind of quality. It seems like the rate of change of flow is a problem and I get a blob in the wall at the start and end of the scarf. In addition (and I likely should create an issue for this) there is a spurious retraction at the end of the scarf joint at each layer (I have retract on layer change off, but it still does a retraction even when there is no travel over the travel limit) - which I think is exacerbating the issue. I'll post an image of a "cube" I just printed, 2 walls, no infill - the leadup to the scarf looks good, you can see a blob at the start of the scarf, then the scarf itself looks okay, then there is a blob followed by underextrusion at the end of the scarf, then the wall leading out of the seam seems to be a little underextruded (that rippley pattern).
I am at a bit of a loss what could be wrong to get such bad results.

![IMG_8938](https://github.com/SoftFever/OrcaSlicer/assets/318118/ba31440f-b57b-4586-93f3-9c68a8442a12)

**Reply by vgdh** [reply]
looks like you have pressure advance problem. Look at the corners.

**Reply by evilDave** [reply]
Okay. That is just about as good as I have ever been able to achieve on the corners... there is a bulge of less than 0.05 mm rounding the corner in the direction of nozzle travel, and almost no bulge on the other corner on most sides... It did have it at 0.6 but was getting really bad underextrusion after the seam so recalibrated to 0.5... any less and the corners are bad, and more leads to underextrusion... Running this test again back at 0.6 linear advance gives the following image.
![IMG_8940](https://github.com/SoftFever/OrcaSlicer/assets/318118/7d9e6b36-1fb3-4ef8-9e41-56513927af71)
I suspect you are correct though, but I am beginning to suspect there is something wrong with my printer causing me to be unable to set the linear advance correctly... Possibly something wrong with the extruder stepper?

Anyway, my comment now is less about the scarf feature - only that that feature in particular is being affected by whatever is wrong with my setup...

**Reply by tsmith35** [reply]
What printer are you using and do you use an AMS (if Bambu)?

**Reply by evilDave** [reply]
It’s an ender 3 v2 neo, no ams just the stock extruder and bowden setup. Running mriscoc firmware (with linear advance obviously, so custom compiled, but nothing else changed). I’ve checked the nozzle and bowden tube and couplings seem fine, esteps are spot on. I’ve even tried very conservative speed and acceleration settings with no real improvement.

**Reply by vgdh** [reply]
Bowden extruders works not so well with this feature.

**Reply by evilDave** [reply]
Oh right, I had not noticed anything throughout the discussion on the scarf joint about it not working on bowden setups… it makes sense though that any changes in flow might limit its effectiveness - I note there is another issue logged to use speed changes rather than flow rate changes (where there is speed overhead) I think that might make it more effective for Bowden. 
I am rebuilding my printer now though, there are too many small extrusion related issues (underexteusion, patterning, z banding, bad corners, dimensional inaccuracy) that I suspect there must be something wrong so hopefully rebuild will turn up something. I’ll update if I do manage something more presentable.

**Comment by markaudacity** [comment]
I can't get Orca to ever actually generate or print a scarf joint, no matter what settings I try, including setting `Scarf around entire wall`. 
Orca 2.2.0, Win11, Bambu X1C

**Reply by systemsNinja** [reply]
SAME for me! its just producing a regular joint.

**Reply by cadriel** [reply]
Count me in on this issue..

**Reply by cadriel** [reply]
maybe related to this https://github.com/SoftFever/OrcaSlicer/issues/7326

**Comment by xantrk** [comment]
Are we the only ones having issues? Surprised this is so stale.

**Comment by DARKFiB3R** [comment]
I had good results with this test but haven't tried it on more challanging models

https://makerworld.com/en/models/422555#profileId-325640
